lib/segment/src/common/mmap_ops.rs

```diff
@@ -41,9 +44,62 @@ pub fn open_write_mmap(path: &Path) -> OperationResult<MmapMut> {
 
     let mmap = unsafe { MmapMut::map_mut(&file)? };
     madvise::madvise(&mmap, madvise::get_global())?;
+
     Ok(mmap)
 }
 
+#[derive(Clone, Debug)]
+pub struct PreheatDiskCache {
+    mmap: Arc<Mmap>,
+    path: Option<PathBuf>,
+}
+
+impl PreheatDiskCache {
+    pub fn new(mmap: Arc<Mmap>, path: Option<impl Into<PathBuf>>) -> Self {
+        Self {
+            mmap,
+            path: path.map(Into::into),
+        }
+    }
+
+    pub fn exec(&self) {
+        preheat_disk_cache(self.mmap.as_ref(), self.path.as_deref());
+    }
+}
+
+fn preheat_disk_cache<T>(mmap: &T, path: Option<&Path>)
+where
+    T: Madviseable + ops::Deref<Target = [u8]>,
+{
+    let separator = path.map_or("", |_| " "); // space if `path` is `Some` or nothing
+    let path = path.unwrap_or(Path::new("")); // path if `path` is `Some` or nothing
```

Comment:
Is `path` only required for debug logs?

Reply:
Only for the logs.

Reply:
let's remove

Reply:
If you insist, but I'd left it as is. Easier to debug if/when something goes wrong, easier to quickly hack some tests in `preheat_disk_cache` (e.g., add time measurements or something) and the path is readily available everywhere in the code.

Reply:
measurements should be moved to telemetery.
Our debug logs is already too verbose